src/core/player.js

```diff
@@ -1,71 +1,126 @@
-import { Assets, Sprite, Application } from "pixi.js";
+import { Assets, Sprite } from "pixi.js";
+import { bulletProjectile } from "./bullet";
 
 export class tankPlayer {
-    constructor(playerX, playerY){
-        // setup the x, y co-ordinates
-        this.playerX = playerX;
-        this.playerY = playerY;
-        this.playerSpeed = 5;
-        this.keys = {}; // dict setup
-    }
+  constructor(playerX, playerY, app) {
+    // setup the x, y co-ordinates
+    this.app = app;
+    this.playerX = playerX;
+    this.playerY = playerY;
+    this.playerSpeed = 5;
+    this.keys = {}; // dict setup
+    this.bullets = [];
+  }
 
-    async initialiseSprite(){
-        // load texture of player, and convert into sprite.
-        const texture = await Assets.load('assets/images/tank.png'); // 'await' keyword used for asynchronous texture loading
-        const sprite = Sprite.from(texture);
-        sprite.anchor.set(0.5, 0.5); // set anchor point to centre of the sprite
-
-        // resizing the texture
-        const desiredWidth = 150;
-        const desiredHeight = 105;
-        sprite.scale.set(desiredWidth / texture.width, desiredHeight / texture.height); // set the scale so there is no sprite distortion
-
-        // initialise x, y to arguements passed through via constructor
-        sprite.x = this.playerX;
-        sprite.y = this.playerY;
-        this.sprite = sprite;
-    }
+  getX() {
+    return this.playerX;
+  }
+
+  getY() {
+    return this.playerY;
+  }
+
+  async createBullet() {
+    const bullet = new bulletProjectile(this.playerX, this.playerY, this.app);
+    await bullet.initialiseSprite();
+    this.app.stage.addChild(bullet.getSprite());
+    this.addBulletToBullets(bullet);
+  }
 
-    addToStage(app) {
-        app.stage.addChild(this.sprite);
+  updateBullets() {
+    for (let i = 0; i < this.getBulletsList().length; i++) {
+      const projectile = this.getBulletsList()[i];
+      projectile.applyGravityToVerticalMotion();
+      projectile.updateBullet();
+
+      // check if bullet has gone off the screen, if it has, then it will be deleted. 
+      if (projectile.getX() > (this.app.canvas.width + 20) || projectile.getX() < 0) {
```

Comment:
Is there motion on the bullets or is it just a line of bullets?

Reply:
simple projectile motion affected by gravity therefore not just a line of bullets. Thing is, in the future, I want the projectile to be affected by air resistance horizontally so that the trajectory of said projectile is not a simple quadratic curve. Should look something like this in the end
<img width="1427" alt="image" src="https://github.com/user-attachments/assets/1a025943-b07a-45ad-9a98-c1630805fed9">

Reply:
But it appears as a stream in the game (not constant stream but still). Is it because the rate of fire is so fast or because the previous bullet isn't being deleted as the next one gets drawn?

Reply:
its because the rate of fire is so fast. A dictionary was essentially used to keep track of what keys the user has pressed. So if I press 'spacebar' (or press and hold), which has a numerical value of 32, in the dictionary it will set it to true (['32'] = true), and keep it at true if the spacebar is held down. If the space bar is let go, the procedure 'keysup()' within the player class will detect this and set the boolean value of '32' to false. As long as ['32'] = true, a new bullet projectile object is instantiated, updated, and therefore drawn onto the screen really quickly. I've made it so that when the bullet goes off the screen it gets deleted. 

- This will be changed to single fire soon though so it will not look like a stream.
- sorry for long paragraph.

Reply:
Oh cool that sounds good